docs/package.nix

```
@@ -0,0 +1,137 @@
+                    #
+                    # This is not perfect, but it seems to cover a vast majority of use cases.
+                    #
+                    # Caveat: even if the package is reached by a different means, the
+                    # path above will be shown and not e.g.
+                    # `${config.services.foo.package}`.
+                    scrubDerivations = namePrefix: pkgSet:
+                      mapAttrs (
+                        name: value: let
+                          wholeName = "${namePrefix}.${name}";
+                        in
+                          if isAttrs value
+                          then
+                            scrubDerivations wholeName value
+                            // optionalAttrs (isDerivation value) {
+                              inherit (value) drvPath;
+                              outPath = "\${${wholeName}}";
+                            }
+                          else value
+                      )
+                      pkgSet;
+                  in {
+                    _module = {
+                      check = false;
+                      args.pkgs = mkForce (scrubDerivations "pkgs" pkgs);
+                    };
+                  }
+                )
+                # avoid having `_module.args` in the documentation
+                {
+                  options = {
+                    _module.args = mkOption {
+                      internal = true;
+                    };
+                  };
+                }
+              ];
+          })
+        )
+        options
+        ;
+
+      transformOptions = opt:
+        opt
+        // {
+          declarations =
+            map (
+              decl:
+                if hasPrefix (toString ../.) (toString decl)
+                then
+                  pipe decl [
+                    toString
+                    (removePrefix (toString ../.))
+                    (removePrefix "/")
+                    (x: {
+                      url = "https://github.com/feel-co/hjem/blob/main/${x}";
+                      name = "<hjem/${x}>";
+                    })
+                  ]
+                else if decl == "lib/modules.nix"
+                then {
+                  url = "https://github.com/NixOS/nixpkgs/blob/master/${decl}";
+                  name = "<nixpkgs/lib/modules.nix>";
+                }
+                else decl
+            )
+            opt.declarations;
+        };
+    })
+    .optionsJSON;
+
+  hjemDocsWeb =
+    pkgs.runCommandLocal "hjem-docs" {
+      nativeBuildInputs = [ndg];
+    } ''
+      mkdir -p $out/share/doc
+
+      # Copy the markdown sources to be processed by ndg
+      cp -rvf ${./inputs} ./inputs
+
+      ndg --verbose html \
+        --jobs $NIX_BUILD_CORES --title "Hjem" \
+        --module-options ${configJSON}/share/doc/nixos/options.json \
+        --manpage-urls ${pkgs.path}/doc/manpage-urls.json \
+        --options-depth 2 \
```

Probably should use 3 so that `hjem.users.<user>` is at the top of the tree. This might turn out weird, not sure exactly, but I suppose when there's a draft render we can see.

```suggestion
        --options-depth 3 \
```